### ISSUE TITLE
AJAX loaded translations not triggered state to re-render components

### ISSUE THREAD

**Issue by volodalexey** [dkfbasel/vuex-i18n#14]
```
export default {
    methods: {
      checkLocaleExists() {
        let {$i18n, $route, $store} = this;
        let locale = $i18n.locale();
        if (!$i18n.exists(locale)) {
          $store.dispatch(actions.dic.FETCH_LOCALE, locale)
            .then((translations) => {
              $i18n.add(locale, translations);
              $i18n.set(locale);
            });
        } else {
          $i18n.set(locale);
        }
      }
    },
    mounted() {
      this.checkLocaleExists();
    }
```

If I load translations by AJAX I can not force component to re-render with new translations.
E.g. when component mounted I have already set locale as 'en', then I check that I do not have translations => download translations => add translations => no re-render.

e.g. part of component that use translations:

```
<template>
    <nav>
        <router-link to="/some">{{ $t('HEADER.MAIN') }}</router-link>
    </nav>
</template>

<script>
  import {mapState} from 'vuex'

  export default {
    computed: {
      ...mapState({
        locale: state => state.i18n.locale
      })
    }
  }
</script>
```

Maybe I have to add some more computed properties? Or how to force re-render my component?

**Comment by tikiatua** [comment]
Hi @volodalexey,

This is an interesting question and definitely a use case that we should support somehow. I will look into this tomorrow and see if I can find a solution.

**Comment by tikiatua** [comment]
Hi there,

Sorry for taking this long time to respond, I got quite swamped at work. We will try to take another look at the problem next week.

**Comment by volodalexey** [comment]
Hi!
Thanks for reply.
For anyone who wants a quick temporary solution:
```
      if (!$i18n.exists(locale)) {
          $i18n.set(undefined); // TODO
          $store.dispatch(actions.dic.FETCH_LOCALE, locale)
            .then((translations) => {
              $i18n.add(locale, translations);
              $i18n.set(locale);
            });
        } else {
          $i18n.set(locale);
        }
      }
```

**Comment by tabun-matadorov** [comment]
I've managed to get component to be updated on store change. I've just override ADD_LOCALE to use `Vue.set()`
```
// Export store
export default {
    namespaced: false,
    ...VuexI18n.store,
    mutations: {
        ...VuexI18n.store.mutations,
        ADD_LOCALE(state, payload) {
            Vue.set(state.translations, payload.locale, payload.translations);
        },
    },
};
```
and that's all.

**Comment by tikiatua** [comment]
Marvelous, I already thought that something like that should work but just did not get around to try it out. I will adapt the code accordingly.

**Comment by tikiatua** [comment]
Ok. This was actually a little bit more difficult. We did not want to use Vue.set in the module, to avoid having to load the Vue module explicitly. 

However we managed to trigger the observer on the translations by calling `state.translations.__ob__.dep.notify();` after updating the store. This seems to do the trick as well. We also added a respective "test case" to the testing page.

Version 1.3.5 - available now - should fix the issue
